src/python/espressomd/visualization_opengl.pyx

```diff
@@ -2016,6 +1878,519 @@ class openGLLive:
+        ax, rx, ry = rotation_helper(self.axis)
+        OpenGL.GL.glRotatef(ax, rx, ry, 0.0)
+
+        n = max(20, self.quality)
+        rotation_angle = np.arctan(
+            (self.radius_1 - self.radius_2) / self.length)
+        l = self.length / np.cos(rotation_angle)
+
+        contour = []
+        for theta in np.linspace(-0.5 * np.pi, 0.5 * np.pi, n):
+            contour.append([np.sin(theta) * 0.5 * self.thickness,
+                            np.cos(theta) * 0.5 * self.thickness + 0.5 * l])
+        for theta in np.linspace(0.5 * np.pi, -0.5 * np.pi, n):
+            contour.append([np.sin(theta) * 0.5 * self.thickness,
+                            -np.cos(theta) * 0.5 * self.thickness - 0.5 * l])
+        contour = np.matmul(np.array(contour),
+                            np.array([[np.cos(rotation_angle), -np.sin(rotation_angle)],
+                                      [np.sin(rotation_angle), np.cos(rotation_angle)]]))
+
+        normals = np.diff(np.array(contour), axis=0)
+        normals /= np.linalg.norm(normals, ord=2, axis=1, keepdims=True)
+        normals = np.roll(normals, 1, axis=1)
+
+        OpenGL.GLE.gleSetJoinStyle(OpenGL.GLE.TUBE_JN_ANGLE)
+        OpenGL.GLE.gleSetNumSides(max(90, 3 * self.quality))
+        OpenGL.GLE.gleSpiral(contour, normals, [0, 0, 1], 0.5 * (self.radius_1 + self.radius_2), 0., 0., 0.,
+                             [[1, 0, 0], [0, 1, 0]], [[0, 0, 0], [0, 0, 0]], 0., 360)
+
+        OpenGL.GLU.gluDeleteQuadric(quadric)
+        OpenGL.GL.glPopMatrix()
+
+
+class SimplePore(Shape):
+    """
+    Drawable Shape SimplePore.
+
+    """
+
+    def __init__(self, shape, particle_type, color, material,
+                 quality, box_l, rasterize_resolution, rasterize_pointsize):
+        super().__init__(shape, particle_type, color, material,
+                         quality, box_l, rasterize_resolution, rasterize_pointsize)
+        self.center = np.array(self.shape.get_parameter('center'))
+        self.axis = np.array(self.shape.get_parameter('axis'))
+        self.length = np.array(self.shape.get_parameter('length'))
+        self.radius = np.array(self.shape.get_parameter('radius'))
+        self.smoothing_radius = np.array(
+            self.shape.get_parameter('smoothing_radius'))
+        self.max_box_l = max(box_l)
+
+    def draw(self):
+        """
+        Draw using OpenGL Extrusion library, if available.
+        Use OpenGL primitives + clip planes, otherwise.
+
+        """
+        set_solid_material(self.color, self.material)
+        OpenGL.GL.glPushMatrix()
+        quadric = OpenGL.GLU.gluNewQuadric()
+        # basic position and orientation
+        OpenGL.GL.glTranslate(self.center[0], self.center[1], self.center[2])
+        ax, rx, ry = rotation_helper(self.axis)
+        OpenGL.GL.glRotatef(ax, rx, ry, 0.0)
+
+        if bool(OpenGL.GLE.gleSpiral):
+            self._draw_using_gle()
+        else:
+            self._draw_using_primitives(quadric)
```

Comment:
You can move the quadric creation and deletion within the body of this `else:` clause. There is no need to waste memory allocation+deallocation on the heap for a large object that is not used.

Reply:
Actually, you can move the quadric creation/deletion directly in the body of `self._draw_using_primitives()`.